### PR TITLE
fix(1857): Show trusted mark on templates/commands list screen

### PR DESCRIPTION
**Pull request by yoshwata**
## Context

This PR fixes the issue below.  
https://github.com/screwdriver-cd/screwdriver/issues/1857

## Objective

The trusted mark is disappeared by the PR below.  
https://github.com/screwdriver-cd/datastore-sequelize/pull/59

That is because the trusted value is assigned only first version of template.
This PR fixes the issue by refer the MAX value of trusted column.
The SQL is like below.
```sql
SELECT
  id,
  name,
  version,
  namespace,
  (
    -- Start the new subquery that this PR add.
    SELECT
      MAX(trusted)
    FROM
      templates
    WHERE
      t1.name = name
      AND t1.namespace = namespace
     -- End the new subquery
  ) AS trusted
FROM
  templates AS t1
WHERE
  t1.id IN (
    SELECT
      MAX(t2.id)
    FROM
      templates AS t2
    WHERE
      t1.name = t2.name
      AND t1.namespace = t2.namespace
      AND t1.id <= t2.id
  )
ORDER BY
  createTime;
```

This PR seems to be temporary because it does not give enough abstracted function to put on `datastore-sequelize`. In other word, that is too much designated for `templates` and `commands` table.

I do not add test because I can't find the way to test that's fits this situation. However, this code is temporary, or which will be deleted one of these days, so I decided not to make the test for such code.

The best idea is that trusted column is divided from templates or commands table in my thought.

## References

https://github.com/screwdriver-cd/screwdriver/issues/1857  
https://github.com/screwdriver-cd/datastore-sequelize/pull/59

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -474,6 +474,27 @@ class Squeakquel extends Datastore {
             findParams.attributes = includedFields.map((field) => {
                 let col = Sequelize.col(field);
 
+                // Temporary treatment to show correct trusted value.
+                // This subQuery is used on fiels of SELECT clause.
+                // This needs to delete after the trusted table generated.
+                if (field === 'trusted') {
+                    // TODO: config.table -> tableName
```
Comment (ibu1224) [inline index.js]: After this [PR](https://github.com/screwdriver-cd/datastore-sequelize/pull/62) merged, this todo will fix.

**Comment by wahapo** [comment]
👍 I think this is the best way to fix the bug as soon as possible.

**Comment by yoshwata** [comment]
I executed the query with our production environment, which have 6450 templates.
```
738 rows in set (0.62 sec)
```

This is the result I execute the query of the [PR](https://github.com/screwdriver-cd/datastore-sequelize/pull/59).
```
738 rows in set (0.44 sec)
```

I found that the query will be slower a little after this PR is merged, but I think that does not become performance issue.

File: index.js (hunk 1 above)
Comment (ibu1224) [reply index.js]: fiexd.

**Comment by yoshwata** [comment]
Dear SDTeam. Could someone review this PR? Let me mention some active or related members @DekusDenial @parthasl 🙇

**Comment by parthasl** [comment]
> Dear SDTeam. Could someone review this PR? Let me mention some active or related members @DekusDenial @parthasl 🙇

@yoshwata Query in the Objective section failed to execute in Postgresql.

**Comment by DekusDenial** [comment]
> > Dear SDTeam. Could someone review this PR? Let me mention some active or related members @DekusDenial @parthasl 🙇
> 
> @yoshwata Query in the Objective section failed to execute in Postgresql.
> ERROR: function max(boolean) does not exist
> LINE 9: MAX(trusted)
> ^
> HINT: No function matches the given name and argument types. You might need to add explicit type casts.
> SQL state: 42883
> Character: 115

@yoshwata I think the `trusted` column needed casting or feel free to migrate the column to numerical type other than bool

**Comment by yoshwata** [comment]
@DekusDenial @parthasl Thanks. I added code to casting `trusted`.

**Comment by parthasl** [comment]
@yoshwata FYI - I don't see much difference in performance

**Pull request merged**